File: Semestr 6/JavaProjekt/Frotnend/src/app/Invoice/invoice.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { BehaviorSubject, Observable } from 'rxjs';
import { Invoice } from '../Models/invoice';
import { app } from '../Core/appip';

@Injectable({
  providedIn: 'root'
})
export class InvoiceService {

  constructor(private http: HttpClient) { }

  public getAll(){
    return this.http.get(`http://${app.ip}/api/v1/invoice`)
  }

  getById(id: number){
    return this.http.get(`http://${app.ip}/api/v1/invoice/`+id)
  }

  public create(body : any){
    return this.http.put(`http://${app.ip}/api/v1/invoice/new`, body)
  }

}

File: Semestr 6/JavaProjekt/Projekt/src/main/java/com/example/projekt/services/InvoiceService.java
package com.example.projekt.services;

import com.example.projekt.models.ApiModels.InvoiceRequest;
import com.example.projekt.models.Invoice;

import java.util.List;

public interface InvoiceService {
    public List<Invoice> getAll();
    public Invoice getById(int id);
    public void create(InvoiceRequest request);
}
